src/Lecture2.hs

```diff
@@ -48,7 +48,13 @@ zero, you can stop calculating product and return 0 immediately.
 84
 -}
 lazyProduct :: [Int] -> Int
-lazyProduct = error "TODO"
+lazyProduct []  = 1
+lazyProduct l   = product l 1
+    where
+        product :: [Int] -> Int -> Int
+        product [] acc      = acc
+        product (0:_) _     = 0
+        product (x:xs) acc  = product xs (acc * x)
```

Comment:
`product` is a name of s standard function from `Prelude` that finds the product of all elements in the list. It works here due to name shadowing. But usually it's better to avoid name conflicts with standard names 🙂